### PR TITLE
Reshaped

### PR DESCRIPTION
**Pull request by KhalidAl-otaibi**
Implemented reshaped #9

### CODE REVIEW

File: Sources/ShapedArray/Operators/Basic.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  Basic.swift
+//  
+//
+//  Created by Khalid Alotaibi on 12/26/22.
+//
```
Comment (rexmas) [inline Sources/ShapedArray/Operators/Basic.swift]: I'll add this to the readme, but we need to follow the same copywrite as in the other files. E.g. https://github.com/swift-numerics-workgroup/swift-shaped-array/blob/99c2f10036a42f66bead022f35d1030d1d39d677/Sources/ShapedArray/ShapedArray.swift#L1-L14

File: Sources/ShapedArray/Core/ShapedArray.swift [hunk 2]
```diff
@@ -22,7 +23,7 @@ public struct ShapedArray<Scalar>: _ShapedArrayProtocol {
     
     /// The dimensions of the array.
     public private(set) var shape: [Int]
-    
+
```
Comment (rexmas) [inline Sources/ShapedArray/Core/ShapedArray.swift]: It seems like you committed unnecessary whitespace in 2 places

**Review by rexmas — CHANGES_REQUESTED** [review]
Another question, why did we include a `Core` directory? we should separately discuss directory structure outside of this pr.

Thanks for doing this, didn't realize it'd be that easy, I'll jump on something else today :)

**Comment by rexmas** [comment]
On a separate topic, i think we should put thought into our commit messages. My go to is to state "what" this does, and "why". Commit history like "Stuff" isn't useful, there's still time to squash all these commits however after adjustments have been made.

**Comment by rexmas** [comment]
Oh, and let's include some unit tests too please

**Comment by JaapWijnen** [comment]
The reshape function is unsafe at the moment I'd suggest adding some asserts to the function to check if the new shape is compatible with the old shape / available data

File: Sources/ShapedArray/Basic.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public func reshaped(to newShape: Int...) -> ShapedArray {
+        return self.reshaped(to: newShape)
+    }
+
+    public func reshaped<T>(like other: ShapedArray<T>) -> ShapedArray {
+        return self.reshaped(to: other.shape)
+    }
+}
```
Comment (rexmas) [inline Sources/ShapedArray/Basic.swift]: files should terminate with a newline

**Review by rexmas — APPROVED** [review]
just add the terminating newline and otherwise looks great!

**Pull request merged**